specs/agents/tracing-spans-compress.md

```diff
@@ -0,0 +1,275 @@
+
+#### Configuration option `span_compression_same_kind_max_duration`
+
+Consecutive spans to the same destination that are under this threshold will be compressed into a single composite span.
+This option does not apply to [composite spans](#composite-span).
+This reduces the collection, processing, and storage overhead, and removes clutter from the UI.
+The tradeoff is that the DB statements of all the compressed spans will not be collected. 
+
+|                |          |
+|----------------|----------|
+| Type           | `duration`|
+| Default        | `5ms`    |
+| Dynamic        | `true`   |
+
+### Composite span
+
+Compressed spans don't have a physical span document.
+Instead, multiple compressed spans are represented by a composite span.
+
+#### Data model
+
+The `timestamp` and `duration` have slightly similar semantics,
+and they define properties under the `composite` context.
+
+- `timestamp`: The start timestamp of the first span.
+- `duration`: gross duration (i.e., _<last compressed span's end timestamp>_ - _<first compressed span's start timestamp>_).
+- `composite`
+    - `count`: The number of compressed spans this composite span represents.
+      The minimum count is 2 as a composite span represents at least two spans.
+    - `sum.us`: sum of durations of all compressed spans this composite span represents in microseconds.
+      Thus `sum.us` is the net duration of all the compressed spans while `duration` is the gross duration (including "whitespace" between the spans).
+    - `compression_strategy`: A string value indicating which compression strategy was used. The valid values are:
+        - `exact_match` - [Consecutive-Exact-Match compression strategy](tracing-spans-compress.md#consecutive-exact-match-compression-strategy)
+        - `same_kind` - [Consecutive-Same-Kind compression strategy](tracing-spans-compress.md#consecutive-same-kind-compression-strategy)
+
+#### Effects on metric processing
+
+As laid out in the [span destination spec](tracing-spans-destination.md#contextdestinationserviceresource),
+APM Server tracks span destination metrics.
+To avoid compressed spans to skew latency metrics and cause throughput metrics to be under-counted,
+APM Server will take `composite.count` into account when tracking span destination metrics.
+
+### Compression algorithm
+
+#### Eligibility for compression
+
+A span is eligible for compression if all the following conditions are met
+- It's an [exit span](tracing-spans.md#exit-spans)
+- The trace context of this span has not been propagated to a downstream service
+- If the span has `outcome` (i.e., `outcome` is present and it's not `null`) then it should be `success`.
+  It means spans with outcome indicating an issue of potential interest should not be compressed.    
+
+The latter condition is important so that we don't remove (compress) a span that may be the parent of a downstream service.
```

Comment:
nit: this is now referring to the middle condition. Should we reorder the conditions or change the wording on this line?

Reply:
```suggestion
The second condition is important so that we don't remove (compress) a span that may be the parent of a downstream service.
```